pallets/loans/src/lib.rs

```diff
@@ -885,12 +902,15 @@ impl<T: Config> Pallet<T> {
     fn total_collateral_asset_value(
         borrower: &T::AccountId,
     ) -> result::Result<FixedU128, Error<T>> {
-        let collateral_assets = AccountCollateralAssets::<T>::get(borrower);
-        if collateral_assets.is_empty() {
-            return Err(Error::<T>::NoCollateralAsset);
-        }
         let mut total_asset_value: FixedU128 = FixedU128::zero();
-        for currency_id in collateral_assets.iter() {
+        for currency_id in Currencies::<T>::get().iter() {
+            if !AccountDeposits::<T>::contains_key(currency_id, borrower) {
```

Comment:
seems we can remove this if and keep only the second as  we can check both `voucher_balance` & `is_collateral` there

Reply:
Utilize the default value when the user doesn't have deposits? Hmm, maybe it's a good idea.